File: csharp/DecoratorPattern/Egg.cs

using System;
namespace DecoratorPattern
{
    /// <summary>
    /// 鸡蛋
    /// </summary>
    public class Egg : Ingredients
    {
        public Egg(Food food)
            : base(food)
        {
            this.Name = "鸡蛋" + this.food.Name;
        }

        public override void Cook()
        {
            Console.WriteLine(this.Name);
        }
    }
}

File: c++/DecoratorPattern/Egg.cpp
#include "stdafx.h"
#include "Egg.h"


Egg::Egg(Food *food) : Ingredients(food)
{
	this->Name.append("¼¦µ°");
	this->Name.append(this->food->Name);
}


void Egg::Cook()
{
	cout<<this->Name<<endl;
}
